### ISSUE TITLE
Error from brew update: Failure while executing: git stash pop --quiet

### ISSUE THREAD

**Issue by JetForMe** [Homebrew/homebrew#44973]
Note that this seems to be different than a similar [issue](https://github.com/Homebrew/homebrew/issues/42488) reported. OS X 10.10.5, not El Cap. I haven't done anything with brew since a week or two ago when you guys helped me resolve an issue with a bad formula I had that was preventing brew update from working. Once we solve that, all seemed well, but now I'm running into this. I haven't attempted to edit or write any formulas since then.

	$ brew install faad
	Error: No available formula for faad 
	==> Searching formulae...
	faad2
	==> Searching taps...
	==> You haven't updated Homebrew in a while.
	A formula for faad might have been added recently.
	Run `brew update` to get the latest Homebrew updates!

	$ brew update
	Stashing your changes:
	 M Library/Formula/dtc.rb
	?? Library/Formula/xively.rb
	Error: Failure while executing: git stash pop --quiet

	$ brew search faad
	faad2
	Error: GitHub API rate limit exceeded for 157.22.42.2. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)
	Try again in 31 minutes 30 seconds, or create an personal access token:
	  https://github.com/settings/tokens
	and then set the token as: HOMEBREW_GITHUB_API_TOKEN

	$ brew install faad2
	==> Downloading https://homebrew.bintray.com/bottles/faad2-2.7.yosemite.bottle.1.tar.gz
	######################################################################## 100.0%
	==> Pouring faad2-2.7.yosemite.bottle.1.tar.gz
	🍺  /usr/local/Cellar/faad2/2.7: 16 files, 748K

	$ brew update
	Stashing your changes:
	UU Library/Formula/dtc.rb
	?? Library/Formula/xively.rb
	Library/Formula/dtc.rb: needs merge
	Library/Formula/dtc.rb: needs merge
	Library/Formula/dtc.rb: unmerged (980532becf3677a2ce0696cccc2f630213fc2596)
	Library/Formula/dtc.rb: unmerged (bed2c95f4fae9ecf6c160ccd37e041c4007930ab)
	Library/Formula/dtc.rb: unmerged (05b82db5f95f5b1cd55702399a90cb9824062e4d)
	fatal: git-write-tree: error building trees
	Cannot save the current index state
	Error: Failure while executing: git stash save --include-untracked --quiet
	
	$ brew doctor
	Your system is ready to brew.

**Comment by DomT4** [comment]
There are some local modifications that can't be stashed and then popped back out for whatever reason. If you've deliberately made changes to those formula move them elsewhere briefly and then:

```
cd $(brew --prefix) && git fetch && git reset --hard origin/master
```

If you haven't deliberately made changes to those formula just run the command above without moving anything. The command wipes out local changes to the git-known files and restores them to match the latest upstream commit here.

**Comment by JetForMe** [comment]
That worked. Sorry, I edited my original post after you answered, but I think what I wrote is moot.

I haven't done anything with brew since a week or two ago when you guys helped me resolve an issue with a bad formula I had that was preventing brew update from working. Once we solve that, all seemed well, but now I'm running into this. I haven't attempted to edit or write any formulas since then.

Thanks!
